engine/execution/migration/badgerpebble_test.go

```diff
@@ -0,0 +1,157 @@
+			transactions := bstorage.NewTransactions(metrics, bdb)
+			collections := bstorage.NewCollections(bdb, transactions)
+			chunkDataPacks := store.NewChunkDataPacks(metrics, pebbleimpl.ToDB(pdb), collections, bstorage.DefaultCacheSize)
+
+			err = headers.Store(genesis)
+			require.NoError(t, err)
+
+			getLatestFinalized := func() (uint64, error) {
+				return genesis.Height, nil
+			}
+
+			// create execution state module
+			es := state.NewExecutionState(
+				nil,
+				commits,
+				nil,
+				headers,
+				collections,
+				chunkDataPacks,
+				results,
+				myReceipts,
+				events,
+				serviceEvents,
+				txResults,
+				db,
+				getLatestFinalized,
+				trace.NewNoopTracer(),
+				nil,
+				false,
+			)
+			require.NotNil(t, es)
+
+			executableBlock := unittest.ExecutableBlockFixtureWithParent(
+				nil,
+				genesis,
+				&unittest.GenesisStateCommitment)
+			header := executableBlock.Block.Header
+
+			err = headers.Store(header)
+			require.NoError(t, err)
+
+			computationResult := testutil.ComputationResultFixture(t)
+			computationResult.ExecutableBlock = executableBlock
+			computationResult.ExecutionReceipt.ExecutionResult.BlockID = header.ID()
+
+			// save execution results
+			err = es.SaveExecutionResults(context.Background(), computationResult)
+			require.NoError(t, err)
+
+			// read the saved results before migration
+			badgerResults, badgerCommits := createStores(badgerimpl.ToDB(bdb))
+			bresult, bcommit, err := readResultsForBlock(
+				header.ID(), badgerResults, badgerCommits)
+			require.NoError(t, err)
+
+			// mock that the executed block is the last executed and sealed block
+			ps := new(protocolmock.State)
+			mockSnapshot := createSnapshot(header)
+			params := new(protocolmock.Params)
+			params.On("SporkID").Return(mainnet26SporkID)
+			ps.On("Params").Return(params)
+			ps.On("AtHeight", mock.Anything).Return(
+				func(height uint64) protocol.Snapshot {
+					if height == header.Height {
+						return mockSnapshot
+					}
+					return invalid.NewSnapshot(fmt.Errorf("invalid height: %v", height))
+				})
+			ps.On("AtBlockID", mock.Anything).Return(
+				func(blockID flow.Identifier) protocol.Snapshot {
+					if blockID == header.ID() {
+						return mockSnapshot
+					} else if blockID == genesis.ID() {
+						return createSnapshot(genesis)
+					}
+					return invalid.NewSnapshot(fmt.Errorf("invalid block: %v", blockID))
+				})
+			ps.On("Sealed", mock.Anything).Return(mockSnapshot)
+
+			// run the migration
+			require.NoError(t, MigrateLastSealedExecutedResultToPebble(unittest.Logger(), bdb, pdb, ps, rootSeal))
+
+			// read the migrated results after migration
+			pebbleResults, pebbleCommits := createStores(pebbleimpl.ToDB(pdb))
+			presult, pcommit, err := readResultsForBlock(
+				header.ID(), pebbleResults, pebbleCommits)
+			require.NoError(t, err)
+
+			// compare the migrated results
+			require.Equal(t, bresult, presult)
+			require.Equal(t, bcommit, pcommit)
+		})
+	})
+}
```

Comment:
It looks like this test covers the case where pebble is uninitialized. can you also add tests for the cases where a previous migration has run and
1. badger has a newer executed block than pebble
2. pebble has a newer executed block than badger

Reply:
Fixed f565cc76d3e